### PR TITLE
Create backup instead of overwriting log files

### LINKED ISSUE #680: Backup logfile before overwriting
## Description

Presently when using `--logfile`, lighthouse overwrites any existing logile. This results in lost logs, which is bad :( 

## Steps to resolve

- When `--logfile` is supplied, check to see if the given file exists. If so, create a backup of it.
    - Avoid causing the same problem as before by overwriting backups. Perhaps add some random suffix to the logfile.

### PR DESCRIPTION
**Pull request by pscott**
## Issue Addressed

Closes #680 

## Proposed Changes

Before creating the log file, we now check for an existing one. If it exists, we rename it with the current timestamp as a suffix.

### CODE REVIEW

File: lighthouse/environment/src/lib.rs [hunk 1]
```diff
@@ -229,6 +231,24 @@ impl<E: EthSpec> Environment<E> {
 
     /// Sets the logger (and all child loggers) to log to a file.
     pub fn log_to_json_file(&mut self, path: PathBuf) -> Result<(), String> {
+        // Creating a backup if the logfile already exists.
+        if path.exists() {
+            let start = SystemTime::now();
+            let timestamp = start
+                .duration_since(UNIX_EPOCH)
+                .expect("Time went backwards")
```
Comment (paulhauner) [inline lighthouse/environment/src/lib.rs]: I think we should return an `Err` here instead of a panic. Unlikely to see it, but it's good hygiene IMO.

Comment (pscott) [reply lighthouse/environment/src/lib.rs]: Yup good point!

**Comment by paulhauner** [comment]
Could you please resolve the merge conflict :)

**Comment by pscott** [comment]
@paulhauner done! :)

**Comment by paulhauner** [comment]
Thank you!

**Pull request merged**